src/alliance_disease_association_ingest/transform.py

```diff
@@ -19,57 +30,63 @@
     "ZFIN": "infores:zfin",
 }
 
-koza_app = get_koza_app("alliance_disease_association")
+predicate_map = {
+    "is_implicated_in": "biolink:contributes_to",
+    "is_marker_for": "biolink:associated_with",
+    "is_model_of": "biolink:model_of"
+}
+
+# Set to keep track of unique associations
+seen_associations = set()
 
 while (row := koza_app.get_row()) is not None:
     subject_category = row["DBobjectType"]
     if subject_category == 'gene':
-        AssociationClass = GeneToDiseaseAssociation
-        predicate = "biolink:related_to"  # TODO: sort out accurate predicates for each kind of row
-    elif subject_category == 'allele':
+        continue  # Skip processing for 'gene'
+
+    #if subject_category == 'gene':
+    #    AssociationClass = GeneToDiseaseAssociation
+    if subject_category == 'allele':
         AssociationClass = VariantToDiseaseAssociation
-        predicate = "biolink:related_to"  # TODO: sort out accurate predicates for each kind of row
     elif subject_category == 'affected_genomic_model':
         AssociationClass = GenotypeToDiseaseAssociation
-        predicate = "biolink:model_of"
     else:
-        # skip this row if there's an association with another kind of entity that we don't yet support, consider logging?
+        koza_app.next_row()
         continue
 
-    ## from Alliance disease association UI:
-    # "Is Implicated in" means that some variant of the gene is shown to function in causing or modifying a disease (for human) or a disease model state.
-    # "Is a marker for" is used when there is evidence of an association but insufficient evidence to establish causality and does not necessarily imply that the existence of, or change in the biomarker is causal for the disease, but rather may result from it.
-
-    ##  predicates map
-    # biomarker_via_orthology
-    # implicated_via_orthology
-    # is_implicated_in
-    # is_marker_for : biolink:biomarker_for
-    # is_model_of : biolink:model_of
-    # is_not_implicated_in
-
-    if row["AssociationType"] == "is_model_of":
-        predicate = "biolink:model_of"
-#    elif row["AssociationType"] == "is_marker_for":
- #       predicate = "biolink:biomarker_for"
-    else:
-        # skip this row if there's an association type that we don't yet support
+    association_type = row["AssociationType"]
+    predicate = predicate_map.get(association_type)
+
+    if predicate is None:
+        koza_app.next_row()
         continue
 
+    unique_id = (row["DBObjectID"], predicate, row["DOID"], row["AssociationType"])
```

Comment:
It would be good to figure out what the difference is on columns with the same DBObjectID, AssociationType, DOID. If it's publications, that might be enough to put pub into the unique key (or sort out how to gather the complete pub list - Koza isn't well suited to that yet). It could be that it's due to environmental conditions, as in, a wild-type genotype + a chemical is producing a disease model, and our plan for that right now will be to exclude non-standard environments - but it might be a bit of work to get the environment info.